### PR TITLE
feat(gtk): add mouse/cursor events

### PR DESCRIPTION
**Pull request by amrbashir**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [X] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Review by wusyong — APPROVED** [review]

File: src/application/event_loop.rs [hunk 1]
```diff
@@ -429,6 +431,115 @@ impl<T: 'static> EventLoop<T> {
               }
               Inhibit(false)
             });
+
+            let tx_clone = event_tx.clone();
+            window.connect_enter_notify_event(move |_, _| {
+              if let Err(e) = tx_clone.send(Event::WindowEvent {
+                window_id: id,
+                event: WindowEvent::CursorEntered {
+                  // FIXME: currently we use a dummy device id, find if we can get device id from gtk
```
Comment (wusyong) [inline src/application/event_loop.rs:440]: Is it possible to `get_device` from closure's parameter and then get either vendor id or product id?

File: src/application/event_loop.rs [hunk 2]
```diff
@@ -429,6 +431,115 @@ impl<T: 'static> EventLoop<T> {
...
+              if let Some(cursor) = display
+                .get_device_manager()
+                .and_then(|device_manager| device_manager.get_client_pointer())
+              {
+                let (_, x, y) = cursor.get_position();
+                if let Err(e) = tx_clone.send(Event::WindowEvent {
+                  window_id: id,
+                  event: WindowEvent::CursorMoved {
+                    position: PhysicalPosition::new(x as f64, y as f64),
+                    // FIXME: currently we use a dummy device id, find if we can get device id from gtk
```
Comment (wusyong) [inline src/application/event_loop.rs:464]: Ditto and the rest below.

File: src/application/event_loop.rs (hunk 1 above)
Comment (amrbashir) [reply src/application/event_loop.rs:440]: yeah it is possible to get the device vendor id or product id but it returns a `GString` and I have no idead what is that and winit expects a `u32`

Comment (wusyong) [reply src/application/event_loop.rs:440]: The field of DeviceID is not public. I think you are free to change to any type you want. Be sure to implement Copy and Clone trait. I know this is usually not recommended but compatibility with winit API is more important to us.

Comment (wusyong) [reply src/application/event_loop.rs:440]: Or we can hash the string into an integer.

Comment (amrbashir) [reply src/application/event_loop.rs:440]: `get_product_id` and `get_vendor_id` seems to be behind a feature flag `#[cfg(any(feature = "v3_16", feature = "dox"))]`

and btw winit is also passing an empty `DeviceID` on some of their backends like windows so I think we should be fine.

**Pull request merged**